app/models/concerns/audit_associations.rb

```diff
@@ -0,0 +1,42 @@
+module AuditAssociations
+  def changes_to_save
+    unless audited_options[:associations].blank?
+      super.merge(associated_changes)
+    else
+      super
+    end
+  end
+
+  private
+
+  def associated_changes
+    associations = Array.wrap(audited_options[:associations])
+    associations.inject({}) do |changes_hash, association_name|
+      association_name = association_name.to_s.singularize
+      association_ids = "#{association_name}_ids"
+
+      if send("#{association_ids}_changed?")
+        association_class = association_name.capitalize.constantize
+        change = send("#{association_ids}_change")
+        change_ids = change.flatten.uniq
```

Comment:
why do we need to flatten?

Reply:
flatten is required beacuse `send("#{association_ids}_change")` returns a 2d array, something like `[[1,2],[3,4]]`